src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -277,12 +279,46 @@ async def _init_repositories(repos: list[GitRepo]) -> dict:
     return description
 
 
+async def _check_if_tag_on_branch(repo_path: str, branch: str, tag: str) -> bool:
+    cmd = [
+        "git",
+        "log",
+        "--tags",
+        "--simplify-by-decoration",
+        '--pretty="format:%ai %d"',
+    ]
+    try:
+        data = await run_cmd_line(cmd, repo_path)
+    except CmdLineError as e:
+        raise RuntimeError(
+            " ".join(cmd), "The command was invalid and the cmd call failed."
+        ) from e
+    if not data:
+        return False
+    if "malformed object name" in data:
+        log.error("Tag does not exist. Aborting!")
+        raise RuntimeError("Tag does not exist. Aborting!")
+    for line in data.split("\n"):
+        if branch in line and tag in line:
+            return True
+    foundBranchInData = sum(1 for i in data.split("\n") if branch in i) > 0
+    foundTagInData = sum(1 for i in data.split("\n") if tag in i) > 0
```

Comment:
Good idea asking chatGPT. For now, since this is tested and works and 2 of the three iterations only occur if an error occured to determine the nature of the error. So will leave as is for now if this ok for you :)

Reply:
well if it returns False it will run 3 times.